### ISSUE TITLE
Nan loss after 70K steps on the pre-training.

### ISSUE THREAD

**Issue by gorka96** [prajdabre/yanmtt#7]
Hi again!

I was able to start training my BART model thanks to your help with the tokenizer, but once on the pre-training process, around 70K steps, I got the following NAN loss on the training:

```
...
76930 2.6496577
76940 2.59109
76950 2.596339
76960 2.735384
76970 2.6265104
76980 nan
76990 nan
Saving the model
Loading from checkpoint
Loading from checkpoint
77000 nan
77010 nan
...
```

I used the following training arguments:

```
python3 pretrain_nmt.py -n 1 -nr 0 -g 2 --model_path models/bart_base \
--tokenizer_name_or_path tokenizers/mbart-bpe50k \
--langs xx --mono_src data/train.xx \
--batch_size 4096 \
--multistep_optimizer_steps 4 \
--num_batches 500000 \
--warmup_steps 16000 \
--encoder_layers 6 \
--decoder_layers 6 \
--max_length 128 \
--encoder_attention_heads 12 \
--decoder_attention_heads 12 \
--decoder_ffn_dim 3072 \
--encoder_ffn_dim 3072 \
--d_model 768 \
--fp16
```

Do you have any idea what could have caused this?
The only hypothesis that come to mind are that --fp16 could have caused this, or that I should use a smaller lr than the default one (lr 1e-3).

Thanks for your time,
Gorka

**Comment by prajdabre** [comment]
Hi,

I had the exact same issue with fp16. One needs to work with smaller learning rates. With 6 layer models, the AMP of pytorch only saves you about 10-20% training time. I recommend using fp32.

**Comment by gorka96** [comment]
Thanks for the fast reply!
I used fp16 thinking more about the GPU ram than increasing the training speed, but it seems that the training fits well on my GPUs even with fp32, so I will just do that. Closing this.
